File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int num = 0;
        int counter = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (counter == 0) {
                num = nums[i];
                counter++;
            } else if (num == nums[i]) {
                counter++;
            } else {
                counter--;
            }
        }
        return num;
    }
};

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)/2
        visited = {}
        for num in nums:
            visited[num]=visited.setdefault(num,0)+1 
        for i in visited:
            if visited[i]>n:
                return i

        